src/components/02_atoms/FileUpload/FileUpload.js

```diff
@@ -67,34 +160,187 @@ class FileUpload extends React.Component {
           },
-          () => this.props.onFileUpload(createdFile),
         );
       }
     };
 
-    // @TODO Handle multipe file uploads
-    reader.readAsArrayBuffer(files[0]);
+    reader.readAsArrayBuffer(file);
   };
 
-  render() {
+  /**
+   * Checks the current file has met all the criteria before
+   * being uploaded to the server. If not then will set the
+   * error state and update the total.
+   * Will check file size and extension.
+   * @param {String} type file type.
+   * @param {Number} size file size.
+   * @param {String} name file name.
+   */
+  checkFile = ({ type, size, name }) => {
+    /* eslint-disable camelcase */
+    const errors = {};
+    const extension = type.split('/')[1]; // <MIME_subtype>
+    const {
+      resetState,
+      props: {
+        inputProps: { max_filesize, file_extensions },
+      },
+    } = this;
+
+    // Check file size
+    if (size > max_filesize) {
+      // TODO: Convert max_filesize to MB
+      errors.size =
+        'The file could not be saved because it exceeds 2 MB, the maximum allowed size for uploads.';
+    }
+
+    // Check file extension
+    if (!file_extensions.includes(extension)) {
+      errors.type = `The image file is invalid or the image type is not allowed. Allowed types: ${file_extensions}.`;
+    }
+
+    // Check if there are errors
+    if (Object.keys(errors).length > 0) {
+      errors.name = `The specified file ${name} could not be uploaded.`;
+
+      // Set the state with error and update total
+      this.setState(
+        prevState => ({
+          total: prevState.total + 1,
+          errors: [...prevState.errors, errors],
+        }),
+        resetState,
+      );
+      return false;
+    }
+
+    return true;
+    /* eslint-enable camelcase */
+  };
+
+  /**
+   * Resets the state if all file/s have been uploaded.
+   */
+  resetState = () => {
+    const { total, filesLength } = this.state;
+    if (total === filesLength) {
+      this.setState({ files: [], filesLength: 0, total: 0 });
+    }
+  };
+
+  /**
+   * Will read the file/s, check if multiple files can be uploaded,
+   * check for errors, if no errors then upload the file.
+   * @param {Object} files Selected files.
+   */
+  readFile = files => {
+    const {
+      setBorder,
+      checkFile,
+      uploadFile,
+      props: {
+        inputProps: { multiple },
+      },
+    } = this;
+
+    this.setState({ filesLength: files.length, errors: [] });
+    setBorder();
+
+    if (multiple) {
+      Object.keys(files).forEach(key => {
+        const file = files[key];
+        if (checkFile(file)) {
+          uploadFile(file);
+        }
+      });
+    }
+
+    // If single file upload, upload the first file from the dropped files
+    if (!multiple && checkFile(files[0])) {
+      uploadFile(files[0]);
+    }
+  };
+
+  render = () => {
+    const {
+      onDrop,
+      onClick,
+      getFiles,
+      onDragOver,
+      onDragLeave,
+      state: { errors },
+      props: {
```

Comment:
Instead of multiple I would propose to look at the cardinality option on the schema. This way we would know whether you should be able to upload multiple files or not. This way we don't even need this parameter :) If cardinality > 1 then we should just always allow to upload multiple files.

Reply:
This has been done using the type option, if array it will be multiple, else single upload.